datacube/virtual/impl.py

```diff
@@ -813,19 +821,20 @@ def reproject_band(band, geobox, resampling, dims, dask_chunks=None):
 
     for tile_index in numpy.ndindex(gt.shape):
         sub_geobox = gt[tile_index]
-        subset_band = band[(...,) + subset_geobox_slices(band.geobox, sub_geobox.extent)].chunk(-1)
+        reproject_roi = compute_reproject_roi(band.geobox, sub_geobox, padding=1)
+
```

Comment:
Right around here I want to do some sanity checks. Should I check that the shape of `reproject_roi.roi_dst` matches that of `sub_geobox`?